common/config/config.go

```diff
@@ -597,6 +601,18 @@ type (
 	YamlNode struct {
 		unmarshal func(out any) error
 	}
+
+	// AsyncWorkflowQueueProvider contains the config for an async workflow queue. Only one field must be set.
+	AsyncWorkflowQueueProvider struct {
+		Type                     string                    `yaml:"type"`
+		AsyncWorkflowQueueConfig *AsyncWorkflowQueueConfig `yaml:",inline"`
+	}
+
+	AsyncWorkflowQueueConfig struct {
+		Kafka *KafkaConfig `yaml:"kafka"`
```

Comment:
we should not use this config, because this config doesn't work for us. You can use it as a placeholder.
But we do need to define a new struct to store the config.

Reply:
this was leftover from prototyping. not used in this PR anyway so removing to avoid confusion